### ISSUE TITLE
preview and generate command fails when adding global route middleware via plugin

### ISSUE THREAD

**Issue by siddharth223437** [nuxt/framework#4547]
### Environment

Operating System: Darwin
Node Version: v17.2.0
Nuxt Version: 3.0.0-27491748.3186841
Package Manager: yarn@1.22.10
Builder: vite
User Config: -
Runtime Modules: -
Build Modules: -

### Reproduction

https://github.com/siddharth223437/nuxt3-route-middleware-issue

### Describe the bug

When i added global route middleware via plugin then yarn preview fails. Its the same error with yarn generate as well. Though yarn dev works fine.

### Logs

```shell
[nuxt] [request error] Cannot read properties of undefined (reading 'push')
  at addRouteMiddleware (./server/chunks/app/server.mjs:2446:36)  
  at ./server/chunks/app/server.mjs:3783:3  
  at fn (./server/chunks/app/server.mjs:647:27)  
  at Object.callAsync (./server/chunks/app/server.mjs:410:19)  
  at callWithNuxt (./server/chunks/app/server.mjs:649:23)  
  at applyPlugin (./server/chunks/app/server.mjs:609:39)  
  at applyPlugins (./server/chunks/app/server.mjs:618:11)  
  at async createNuxtAppServer (./server/chunks/app/server.mjs:4032:7)  
  at async renderToString (./server/chunks/handlers/renderer.mjs:268:19)  
  at async ./server/chunks/handlers/renderer.mjs:602:20
```

**Comment by danielroe** [comment]
this should be resolved upstream in nitropack 0.3.8.

Would you try upgrading with `npm upgrade nuxt` or `yarn upgrade nuxt`?

Let me know if not and we can reopen.

**Comment by siddharth223437** [comment]
This is working as expected. Thanks
